test/actions/FilterActionsTest.js

```diff
@@ -0,0 +1,180 @@
+        const unknownEventTypeId = 10000
+
+        const referenceData = {0: {id: 0, name: 'testZero'}, 1: {id: 1, name: 'testOne'}}
+
+        const expectedEventType = { id: 0, name: 'testZero' }
+        const unknownEventType = {id: 10000, name: 'unknown'}
+
+        it('should return a matching object when given a known id', function() {
+            const result = filterActions.findEventTypeInRef(testEventTypeId, referenceData)
+            expect(result).to.deep.equal(expectedEventType)
+        })
+        it('should return a new object with name "unknown" when given an unknown id', function() {
+            const badResult = filterActions.findEventTypeInRef(unknownEventTypeId, referenceData)
+            expect(badResult).to.deep.equal(unknownEventType)
+        })
+    })
+
+    describe('changeEventFilter', function() {
+        const history = []
+        const filter = {ticketId: 0, eventTypes: [1,2]}
+
+        const result = filterActions.changeEventFilter(history)(filter)
+
+        const expectedValue = {type: 'CHANGE_EVENT_FILTER', filter: {ticketId: 0, eventTypes: [1, 2]}}
+        
+        it('should return an object with a type and a filter', function() {
+            expect(result).to.deep.equal(expectedValue)
+        })
+    })
+
+    describe('applyFilter', function() {
+        const history = []
+        const oldFilter = {incidentId: 0}
+        const goodNewFilter = {incidentId: 0}
+        const badNewFilter = {foo: 'bar'}
+
+        const dispatch = function(arg) {
+            return arg
+        }
+        const errorMessage = 'Need to filter on incidentId!'
+        const expectedReturn = { type: 'CHANGE_EVENT_FILTER', filter: {incidentId: 0} }
+
+        it('should throw an error when new filter has no incident id', function() {
+            expect(() => filterActions.applyFilter(history)(oldFilter, badNewFilter)(dispatch)).to.throw(errorMessage)
+        })
+    })
+
+    describe('serializeEventTypesForQuery', function() {
+        const eventTypes = [1, 2]
+        const noEventTypes = []
+
+        const expectedResult = 'eventTypes=1&eventTypes=2'
+
+        const result = filterActions.serializeEventTypesForQuery(eventTypes)
+
+        it('should return a serialized array', function() {
+            expect(result).to.equal(expectedResult)
+        })
+        it('should return an empty string if given an empty array', function() {
+            const emptyStringResult = filterActions.serializeEventTypesForQuery(noEventTypes)
+            expect(emptyStringResult).to.equal('')
+        })
+    })
+
+    describe('serializeFiltersForUrl', function() {
+        const filterNoEventTypes = { incidentId: 0 }
+        const filterWithEventTypes = {incidentId: 0, eventTypes: [1, 2]}
+        const filterWithEverything = {incidentId: 0, eventTypes: [1, 2], foo: 'bar'}
+
+        const expectedFilterTokensWithoutEventTypes = '?'
+        const expectedFilterTokensWithEventTypes = '?eventTypes=1&eventTypes=2'
```

Comment:
Use describe blocks to isolate the variable definitions to the scope of the tests in which their values are consumed.